hydromt_fiat/workflows/exposure_vector.py

```diff
@@ -957,19 +958,21 @@ def setup_ground_elevation(
                 exposure_geoms=self.get_full_gdf(self.exposure_db),
             )
             # Unit conversion
-            if (unit == "meters" or unit == "m") and (
-                self.unit == "feet" or self.unit == "ft"
+            if (unit == Units.meters.value) and (
+                self.unit == Units.feet.value
             ):
                 self.exposure_db["Ground Elevation"] = self.exposure_db[
                     "Ground Elevation"
                 ].apply(lambda x: x * 3.28084)
 
-            elif (unit == "feet" or unit == "ft") and (
-                self.unit == "meters" or self.unit == "m"
+            elif (unit == Units.feet.value) and (
+                self.unit == Units.meters.value
             ):
                 self.exposure_db["Ground Elevation"] = self.exposure_db[
                     "Ground Elevation"
                 ].apply(lambda x: x / 3.28084)
+            elif (unit == Units.feet.value) and (self.unit == Units.feet.value) or (unit == Units.meters.value) and (self.unit == Units.meters.value):
+                pass
             else:
                 self.logger.warning(
                     "The elevation unit is not valid. Please provide the unit of your ground elevation in 'meters' or 'feet'"
```

Comment:
Nice, already much better! 

suggestion to remove the last conditional that checks if the units are the same, and then explicitely does nothing.
Useful trick for cleaning up code like that is to see if you can invert the condition:
Instead of checking at the end if the units are equal, check at the start if they are not.
```
if unit != self.unit:
    if unit == Units.meters.value and self.unit == Units.feet.value:
        self.exposure_db["Ground Elevation"] = self.exposure_db[
            "Ground Elevation"
        ].apply(lambda x: x * 3.28084)
    elif unit == Units.feet.value and self.unit == Units.meters.value:
        self.exposure_db["Ground Elevation"] = self.exposure_db[
            "Ground Elevation"
        ].apply(lambda x: x / 3.28084)
```

a final suggestion is to remove so called magic numbers from the code, which also makes code harder to understand. You could add the conversion factors to an enum, use that enum in a class (like Unitfulvalue in floodadapt) to handle conversion, or just use that enum here directly

Reply:
nice suggestion! Made the changes